### ISSUE TITLE
Secure hash

### ISSUE THREAD

**Issue by haroldLuiz** [rathena/rathena#8042]
### Is your request related to a missing official feature?

No

### Describe the solution you'd like

Having plain text passwords or md5 passwords are a real security issue since, lets be real, most people reuse passwords.

Adding a last layer on the emulator to hash all login passwords would be great!
What if on the last steps when saving a login password on the DB and on the last steps when comparing a login password on the DB, you just always did:

"For legacy systems using [bcrypt](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#bcrypt), use a work factor of 10 or more and with a password limit of 72 bytes."
https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html

I may be wrong, but i dont think this would be hard to change and would be a great security addition!

**Comment by Lemongrass3110** [comment]
Related to #3626 and #7650

**Comment by haroldLuiz** [comment]
> Related to #3626 and #7650

True, but i believe implementing this just on the server would be a great first step, keep all the network stack the same, all the application the same, but whenever something will check if "X" is a password on the DB, bcrypt it first, whenever something will save "X" as a password on the DB, bcrypt it first.

Having passwords not actually leaked in a useful way when a server is hacked somehow would be great.
